app/controllers/api/v1/profile/mfa_controller.rb

```diff
@@ -0,0 +1,74 @@
+class Api::V1::Profile::MfaController < Api::BaseController
+  before_action :set_user
+  before_action :check_mfa_feature_available
+  before_action :set_mfa_service
+  before_action :check_mfa_enabled, only: [:destroy, :backup_codes]
+  before_action :check_mfa_disabled, only: [:create, :verify]
+  before_action :validate_otp, only: [:verify, :backup_codes, :destroy]
+  before_action :validate_password, only: [:destroy]
+
+  def show; end
+
+  def create
+    @mfa_service.enable_two_factor!
+  end
+
+  def verify
+    @backup_codes = @mfa_service.verify_and_activate!
+  end
+
+  def destroy
+    @mfa_service.disable_two_factor!
+  end
+
+  def backup_codes
+    @backup_codes = @mfa_service.generate_backup_codes!
+  end
+
+  private
+
+  def set_user
+    @user = current_user
+  end
+
+  def set_mfa_service
+    @mfa_service = Mfa::ManagementService.new(user: @user)
+  end
```

Comment:
Nitpick, we can use this approach, which is more idiomatic. It memoizes the service instance and eliminates the need for a separate `before_action` callback

```
  def mfa_service
    @mfa_service ||= Mfa::ManagementService.new(user: @user)
  end
```